gramps/plugins/lib/libprogen.py

```diff
@@ -925,7 +914,13 @@ def __get_or_create_citation(self, source_title, date_text, page_text='', page_r
                 sattr.set_value(attr_text)
                 citation.add_attribute(sattr)
 
-        self.dbase.commit_citation(citation, self.trans)
+            self.dbase.add_citation(citation, self.trans)
+            self.dbase.commit_citation(citation, self.trans)
```

Comment:
It is not necessary to commit after an add; the db add_citation does the commit_citation for you (after adding Gramps_ID and handle).  A bit better performance if you don't do both.